File: app/src/main/java/io/github/howiezuo/unsplash/feature/main/MainComponent.java
package io.github.howiezuo.unsplash.feature.main;

import dagger.Component;
import io.github.howiezuo.unsplash.app.ActivityScope;

@ActivityScope
@Component(
        modules = {
                MainViewModelModule.class
        }
)
public interface MainComponent {

    void inject(MainActivity activity);

}

File: app/src/main/java/io/github/howiezuo/unsplash/feature/main/MainComponent.kt
package io.github.howiezuo.unsplash.feature.main

import dagger.Component
import io.github.howiezuo.unsplash.app.ActivityScope


@ActivityScope
@Component(
        modules = arrayOf(
                MainViewModelModule::class
        )
)
interface MainComponent {

    fun inject(activity: MainActivity)

}
